### DISCUSSION TITLE
Difference in virial data from cp2k `.log` file and `.stress` file

### DISCUSSION THREAD

**Discussion by PWFYeung** [deepmodeling/dpdata#322] [General]
**Summary**

The virial outputs in the `.log` file and the `.stress` file by CP2K are different, but dpdata (fmt="cp2k/output") only reads the `.log` file. However, CP2K seems to output `pv_virial` in the `.log` file but print the `pv_total` in the `.stress` file. This might be problematic for generating virial data using `dpdata`.

**Details**
- [CP2K virial_methods.F](https://github.com/cp2k/cp2k/blob/master/src/virial_methods.F) 

**STRESS TENSOR FROM `.log` FILE OF ONE SINGLE FRAME CALCULATION**
```text
  STRESS TENSOR [GPa]

            X               Y               Z
  X      -3.22841220      0.03875193      0.35205443
  Y       0.03875193     -2.99491119     -0.26930668
  Z       0.35205443     -0.26930668     -3.16014758

  1/3 Trace(stress tensor):  -3.12782366E+00

  Det(stress tensor)      :  -2.99521219E+01


```

**`.stress` FILE OF THE SAME CALCULATION**
```text
#   Step   Time [fs]            xx [bar]            xy [bar]            xz [bar]            yx [bar]            yy [bar]            yz [bar]            zx [bar]            zy [bar]            zz [bar]
       0       0.000   -27535.1132302270      252.2330873873     3489.2605728168      252.2330873873   -25741.2101685947    -2388.1540627668     3489.2605728168    -2388.1540627668   -26927.8388906513

```

**Comment by wanghan-iapcm** [comment]
`dpdata` requires virial. 
the stress may include kinetic contribution, which makes it different from virial.

**Reply by PWFYeung** [reply]
Thank you for your reply.

After digging into the issue, it seems that CP2K uses `pv_total` to change the box size. Also, LAMMPS appears to include the kinetic contribution when calculating the pressure of the system. There is also a paper explicitly saying that including the kinetic contribution gives more accurate results. Would it be possible that `pv_total` is a more ideal choice, or is there a reason for using `pv_virial`? Please do correct me if I'm wrong. I hope I am not causing any trouble.

**References**
- [CP2K cell_opt_utils.F](https://github.com/cp2k/cp2k/blob/master/src/motion/cell_opt_utils.F) _Keyword: ptens_
- [CP2K gopt_f77_methods.F](https://github.com/cp2k/cp2k/blob/master/src/motion/gopt_f77_methods.F) _Keyword: ptens_
- [LAMMPS Compute Pressure](https://docs.lammps.org/compute_pressure.html)
- [The Paper Discussing Virial](https://doi.org/10.1016/j.ijsolstr.2008.03.016)
